### PR TITLE
DEFEDIT-4937 Refactor defold-project-search-test to prevent timeout failures

### PR DESCRIPTION
**Pull request by matgis-king**
### Technical changes
* All the match tests have been merged into a single test so that the `save-data` future is produced once and shared among the tests.
* The time it takes to produce the `save-data` is excluded from the timeout check, and the timeout has been lowered as a result.
* Moved search-specific test files from the regular test project to the `search_project`.
* Got rid of a potential contentious `swap!` interaction where the consumer thread fought against anything touching the future.

### CODE REVIEW

File: editor/test/editor/defold_project_search_test.clj [hunk 1]
```diff
@@ -8,69 +8,77 @@
...
-           consumer))
-  nil)
-
-(defn- consumer-started? [consumer-atom]
-  (-> consumer-atom deref :future some?))
-
-(defn- consumer-finished? [consumer-atom]
-  (if-let [pending-future (-> consumer-atom deref :future)]
+  {:consumed-atom (atom [])
+   :future-atom (atom nil)
```
Comment (matgis-king) [inline editor/test/editor/defold_project_search_test.clj:18]: Broke up into two separate atoms to ensure there is no `swap!` retry contention. Now only the consumer thread invokes `swap!`.

File: editor/test/editor/defold_project_search_test.clj [hunk 2]
```diff
@@ -119,122 +127,115 @@
     (let [pattern (project-search/compile-find-in-files-regex "fOoO")]
       (is (= "fooo" (re-matches pattern "fooo"))))))
 
-(deftest make-file-resource-save-data-future-test
```
Comment (matgis-king) [inline editor/test/editor/defold_project_search_test.clj:122]: From this point on it is mostly indentation, apart from each test case now using the shared `save-data-future`.

File: editor/test/editor/defold_project_search_test.clj [hunk 3]
```diff
@@ -8,69 +8,77 @@
...
+(def ^:const timeout-ms 1000)
 
 (defn- make-consumer [report-error!]
-  (atom {:consumed [] :future nil :report-error! report-error!}))
-
-(defn- consumer-start! [consumer-atom results-fn!]
-  (swap! consumer-atom
-         (fn [consumer]
-           (when (:future consumer)
-             (future-cancel (:future consumer)))
```
Comment (vlaaad) [inline editor/test/editor/defold_project_search_test.clj:23]: Woah, that was really dangerous — side-effects in a retriable block!

**Review by vlaaad — APPROVED** [review]
LGTM!

File: editor/test/editor/defold_project_search_test.clj (hunk 3 above)
Comment (vlaaad) [reply editor/test/editor/defold_project_search_test.clj:23]: It actually wasn't — `future-cancel` is idempotent.

Comment (matgis-king) [reply editor/test/editor/defold_project_search_test.clj:23]: Yep. That tripped me up as well! 🙂

**Pull request merged**